RTech.CoffeeMachine.Api.Tests/BrewCoffee/Controllers/CoffeeMachineControllerTests.cs

```diff
@@ -120,29 +113,110 @@ public async Task GetBrewStatus_WhenCoffeeReady_ShouldReturn200()
         dateTimeProvider.Setup(p => p.GetLocalNow())
             .Returns(nowLocal);
 
-        var response = await GetEndpointResponse(services =>
-        {
-            services.AddSingleton<IDateTimeProvider>(dateTimeProvider.Object);
-        });
+        var response = await GetEndpointResponse(dateTimeProvider: dateTimeProvider.Object);
 
         response.StatusCode.ShouldBe(HttpStatusCode.OK);
         var content = await response.Content.ReadFromJsonAsync<BrewStatusResponse>();
         content.ShouldNotBeNull();
-        content.Message.ShouldBe("Your piping hot coffee is ready");
+        content.Message.ShouldBe(BREW_READY_HOT_MESSAGE);
 
         var offsetStr = DateTime.Now.ToString("zzz").Replace(":", "");
         var expectedPrepared = $"2023-01-02T03:04:05{offsetStr}";
         content.Prepared.ShouldBe(expectedPrepared);
     }
 
-    private Task<HttpResponseMessage> GetEndpointResponse(Action<IServiceCollection> servicesConfiguration)
+    [Fact]
+    public async Task GetBrewStatus_WhenTemperatureRequestThrowsException_ShouldReturn502()
```

Comment:
This is the start of new integration tests, to cover the new "iced coffee" message logic.